### PR TITLE
k8s: Rely on the USING_NFD environment variable passed by the jobs

### PR DESCRIPTION
**Pull request by fidencio**
Let's make sure we can rely on the tests passing down whether they want to be tested using Node Feataure Discovery or not.

Right now, only the TDX job has this option set to "true", all the other jobs have this option set to "false".

We can and have to merge this one before merging the NFD related patches as:
1) It causes no harm in exporting this environment variable, but not
   having it used
2) It will allow us to test the NFD after this one is merged, as changes
   in the yaml file, in the case of the pull_request_target event,  are
   not taken into consideration before they're merged

Fixes: #7495

### CODE REVIEW

**Comment by fidencio** [comment]
There's no need to have `ok-to-test` set as the changes in the yaml file are not being taken into consideration before it's merged.  We're setting the `force-skip-ci` label as we're not doing any changes related to the code itself, only to the tests yaml files.

**Review by zvonkok — APPROVED** [review]
LGTM, would be great if the other TEEs follow this pattern.

**Review by jepio — APPROVED** [review]

**Pull request merged**
